clients/roscpp/src/libros/service_server_link.cpp

```diff
@@ -262,6 +266,7 @@ void ServiceServerLink::callFinished()
 
     current_call_->finished_ = true;
     current_call_->finished_condition_.notify_all();
+    current_call_->call_finished_ = true;
```

Comment:
it looks reasonable to change `call_finished_` into true once current call in the queue is finished.

but is this supposed to be moved from [caller thread](https://github.com/ros/ros_comm/blob/089dd64a6c2d7769d2d8d3b10808eec621a3acf3/clients/roscpp/src/libros/service_server_link.cpp#L374)? adding `call_finished_ = true` here means, it should check if `call_finished_` is true on [caller thread](https://github.com/ros/ros_comm/blob/089dd64a6c2d7769d2d8d3b10808eec621a3acf3/clients/roscpp/src/libros/service_server_link.cpp#L374) instead.

Reply:
> but is this supposed to be moved from caller thread?

It should keep here. There are some cases, such as,
https://github.com/ros/ros_comm/blob/089dd64a6c2d7769d2d8d3b10808eec621a3acf3/test/test_roscpp/test/src/service_call.cpp#L97-L109
in this case, `ServiceServerLink::callFinished` will not be called, and then if ServiceServerLink exit,

ServiceServerLink::~ServiceServerLink -> ... -> ServiceServerLink::cancelCall
https://github.com/ros/ros_comm/blob/089dd64a6c2d7769d2d8d3b10808eec621a3acf3/clients/roscpp/src/libros/service_server_link.cpp#L73-L85

stuck in `while (!local->call_finished_)` if we remove https://github.com/ros/ros_comm/blob/089dd64a6c2d7769d2d8d3b10808eec621a3acf3/clients/roscpp/src/libros/service_server_link.cpp#L374


> adding call_finished_ = true here means, it should check if call_finished_ is true

Okay